### DISCUSSION TITLE
c++ with nodejs

### DISCUSSION THREAD

**Discussion by Veercodeprog** [ParanoidUser/yolo#539] [Galaxy Trivia] [answered]
Any good resources to get started in c++ with nodejs to build libraries?

**Comment by thuongtruong1009** [comment, marked as answer]
Yes, there are several resources that can help you get started with building C++ libraries for Node.js. Here are some options:

Node.js Addons documentation: This is the official documentation provided by Node.js that explains how to build C++ addons for Node.js. It covers everything from setting up your development environment to writing and compiling C++ code, and also provides examples.

Nan: Nan is a popular library for writing Node.js addons in C++. It provides a simple and consistent API that abstracts away many of the platform differences between different versions of Node.js. Nan also has excellent documentation and a large community of users.

node-gyp: node-gyp is a tool for compiling native addon modules for Node.js. It uses a common set of build commands to generate platform-specific files, which makes it easy to create addons that work across different platforms. node-gyp also has detailed documentation and a large community of users.

CMake: CMake is a cross-platform tool for building C++ code that can be used to build Node.js addons. It provides a powerful scripting language that can automate complex build processes and generate platform-specific build files. CMake also has excellent documentation and a large community of users.

Learning C++: If you are new to C++, you may want to start by learning the basics of the language. There are many resources available online, including tutorials, books, and online courses. Some popular options include the C++ documentation, the C++ Primer book, and the LearnCpp.com website.

With these resources, you should be able to get started building C++ libraries for Node.js in no time!
